src/events.rs

```diff
@@ -155,16 +157,23 @@ impl EventNotifier {
 /// events that have happened.
 pub struct EventBufferCollector {
     events: Arc<Mutex<Vec<Event>>>,
+    on_push: mpsc::UnboundedReceiver<()>,
```

Comment:
I'm a bit unclear if this `UnboundedReceiver` is routinely going to be pushed to when it's not being read from. If that ever happens, it might be better to use a [`broadcast`](https://docs.rs/tokio/0.2.22/tokio/sync/broadcast/index.html) channel. Broadcast is technically for use with multiple producers and multiple consumers, but it also has the nice property of being able to have a fixed buffer size, _and_ a `send` function that doesn't ever block. If the buffer gets full and more messages are sent, the oldest messages are simply dropped, which seems likely to be the correct behavior in this case.  And that way you can't just ever consume memory without bound like the unbounded channel will.